### PR TITLE
Update widgets

### PR DESCRIPTION
**Pull request by thellmund**
This pull request fixes some bugs in our widgets and makes some design adjustments.

### CODE REVIEW

**Comment by thellmund** [comment]
@kordianbruck @pfent @jacqueline8711 We have widgets for MVV, calendar events and the cafeteria. Do you think there’s still a need for the cards widget or can it be removed? It’s broken and provides little value for the user, as most of its content can be viewed via dedicated widgets.

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -1,264 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pfent) [inline .idea/codeStyleSettings.xml]: please don't remote the files in .idea

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/widget/TimetableWidgetService.java [hunk 2]
```diff
@@ -112,14 +112,14 @@ public RemoteViews getViewAt(int position) {
             DateFormat timeFormat = android.text.format.DateFormat.getTimeFormat(applicationContext);
             String time = timeFormat.format(currentItem.getStartTime()
                                                        .getTime());
-            time += "-" + timeFormat.format(currentItem.getEndTime()
+            time += " – " + timeFormat.format(currentItem.getEndTime()
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/widget/TimetableWidgetService.java]: A en dash. Nice!
having the format string in strings.xml would probably be better, but this should be fine for now

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/widget/MensaRemoteViewFactory.java [hunk 3]
```diff
@@ -45,32 +46,27 @@ public void onDestroy() {
...
         rv.setTextViewText(R.id.menu_content, menuContent + " (" + currentItem.getTypeShort() + ")");
 
-        String price = CafeteriaPrices.INSTANCE.getPrice(applicationContext, currentItem.getTypeLong());
-        if (price == null) {
-            price = "____";
-        }
-
+        String price = CafeteriaPrices.INSTANCE.getPrice(mApplicationContext, currentItem.getTypeLong());
+        rv.setViewVisibility(R.id.menu_price, price == null ? View.INVISIBLE : View.VISIBLE);
         rv.setTextViewText(R.id.menu_price, price + " €");
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/widget/MensaRemoteViewFactory.java]: maybe put all these in an `if (price != null)`

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/widget/MensaRemoteViewFactory.java [hunk 4]
```diff
@@ -45,32 +46,27 @@ public void onDestroy() {
...
             return null;
         }
-        RemoteViews rv = new RemoteViews(applicationContext.getPackageName(), R.layout.mensa_widget_item);
+
+        RemoteViews rv = new RemoteViews(mApplicationContext.getPackageName(), R.layout.mensa_widget_item);
 
         String menuContent = COMPILE.matcher(currentItem.getName())
                                     .replaceAll("")
                                     .trim();
         rv.setTextViewText(R.id.menu_content, menuContent + " (" + currentItem.getTypeShort() + ")");
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/widget/MensaRemoteViewFactory.java]: uuhh, strings resources would probably be nice

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/widget/MVVWidget.java [hunk 5]
```diff
@@ -140,6 +140,8 @@ static void updateAppWidget(Context context, AppWidgetManager appWidgetManager,
         RemoteViews rv = new RemoteViews(context.getPackageName(), R.layout.mvv_widget);
         rv.setTextViewText(R.id.mvv_widget_station, widgetDepartures.getStation());
 
+        // TODO Offline layout
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/widget/MVVWidget.java]: Is this still a TODO you want to do with this PR?

File: .idea/codeStyleSettings.xml (hunk 1 above)
Comment (thellmund) [reply .idea/codeStyleSettings.xml]: That also occurred in another branch. I did not consciously remove them, but I’ll keep a better eye on what files are being committed in the future.

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/widget/MensaRemoteViewFactory.java (hunk 4 above)
Comment (thellmund) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/widget/MensaRemoteViewFactory.java]: Fixed.

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/widget/MensaRemoteViewFactory.java (hunk 3 above)
Comment (thellmund) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/widget/MensaRemoteViewFactory.java]: Fixed.

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/widget/TimetableWidgetService.java (hunk 2 above)
Comment (thellmund) [reply app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/widget/TimetableWidgetService.java]: Fixed.

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/widget/MVVWidget.java (hunk 5 above)
Comment (thellmund) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/transportation/widget/MVVWidget.java]: Forgot to remove that comment. Fixed now.

**Comment by thellmund** [comment]
Basically done. Waiting until Friday to test on a Samsung device, on which the widgets weren’t displayed correctly.

File: .idea/codeStyleSettings.xml (hunk 1 above)
Comment (pfent) [reply .idea/codeStyleSettings.xml]: @thellmund !ping
please revert the changes to .idea

**Comment by kordianbruck** [comment]
@pfent ready to merge now?

**Comment by thellmund** [comment]
@pfent Please give it another review – I’ve added a handful of commits.

**Comment by pfent** [comment]
Looks good to me, those are mostly cosmetic changes anyways.

@kordianbruck, should we do a point release before merging? I'd consider this PR bugfixes, but what do you think?

**Comment by thellmund** [comment]
@pfent @kordianbruck Don’t include this in the next version (1.6.2). I want to first deploy this in a beta to see if there are any problems.

**Comment by kordianbruck** [comment]
@pfent release done and we now have a 1.6.X release branch. Master is for our next major release 2.X

**Pull request merged**